### PR TITLE
Do null check before matching location, default to "us-central1". Fixes #1208

### LINKED ISSUE #1208: An unexpected error has occurred on deploy
Just making firebase --deploy as usual and got this error. Update to latest version of firebase-tools but still the same error. What could be wrong?

```
[debug] [2019-04-21T11:45:53.781Z] TypeError: Cannot read property 'match' of undefined
    at exports.getAppEngineLocation (/usr/local/lib/node_modules/firebase-tools/lib/functionsConfig.js:41:27)
    at module.exports (/usr/local/lib/node_modules/firebase-tools/lib/deploy/functions/release.js:83:29)
    at _chain (/usr/local/lib/node_modules/firebase-tools/lib/deploy/index.js:22:40)
    at /usr/local/lib/node_modules/firebase-tools/lib/deploy/index.js:69:16
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7)
[error] 
[error] Error: An unexpected error has occurred.
```

ps: I'm trying to move from Node 6 to Node 8

### PR DESCRIPTION
**Pull request by mbleigh**
This is a potential/probable fix for #1208. It seems that some customers do not have a default app engine location but still have functions. This covers the case when `config.cloudResourceLocation` is undefined and also defaults it to `us-central1` when that's the case.

I'm not 100% sure that this is the correct behavior as the interactions between the various bits here are a little hard to follow.

### CODE REVIEW

File: src/deploy/functions/release.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ module.exports = function(context, options, payload) {
   var projectId = context.projectId;
   var sourceUrl = context.uploadUrl;
   var appEngineLocation = getAppEngineLocation(context.firebaseConfig);
-  utils.logWarning("appEngineLocation " + appEngineLocation);
+  logger.debug(`appEngineLocation: ${appEngineLocation}`);
```
Comment (mbleigh) [inline src/deploy/functions/release.js]: I don't know why this was a `logWarning` as it seems like it shouldn't have been. Changed it to a debug, but it might be better to remove altogether? Was this just a stray log that wasn't caught at review time?

Comment (ryanpbrewster) [reply src/deploy/functions/release.js]: as a user this is a pretty confusing message

Comment (joehan) [reply src/deploy/functions/release.js]: Pretty sure its a stray log, we should remove.

**Comment by joehan** [comment]
I'm not sure whether this actually fixes 1208 - are we sure that this is only occurring for users with projects in us-central1?

**Comment by mbleigh** [comment]
@joehan from the error that's being thrown, `appEngineLocation` is `undefined` but getting `match()` called on it. My change in line 50 addresses that, which I'm guessing isn't the result of people in `us-central1` but is the result of people without an app engine location.

**Comment by joehan** [comment]
@bleigh, Yep, this will definitely fix the errors due to calling match() on undefined. However, if a user has appEngineLocation undefined, and they try to deploy to a project with a Firestore location other than us-central1, the deployment will still fail with this fix
